traffic_ops/app/lib/API/Deliveryservice.pm

```diff
@@ -480,6 +480,12 @@ sub safe_update {
 		return $self->not_found();
 	}
 
+	my $tenant_utils = Utils::Tenant->new($self);
+	my $tenants_data = $tenant_utils->create_tenants_data_from_db();
+	if (!$tenant_utils->is_ds_resource_accessible($tenants_data, $ds->tenant_id)) {
```

Comment:
this is not quite right. i think the test needs to be 

if (use_tenancy() and !$tenant_utils->is_ds_resource_accessible($tenants_data, $ds->tenant_id))